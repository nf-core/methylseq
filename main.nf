#!/usr/bin/env nextflow
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    nf-core/methylseq
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Github : https://github.com/nf-core/methylseq

    Website: https://nf-co.re/methylseq
    Slack  : https://nfcore.slack.com/channels/methylseq
----------------------------------------------------------------------------------------
*/

nextflow.enable.dsl = 2

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    GENOME PARAMETER VALUES
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')
params.fasta_index = WorkflowMain.getGenomeAttribute(params, 'fasta_index')
params.bismark_index = WorkflowMain.getGenomeAttribute(params, 'bismark')
params.bwa_meth_index = WorkflowMain.getGenomeAttribute(params, 'bwa_meth')

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    VALIDATE & PRINT PARAMETER SUMMARY
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

WorkflowMain.initialise(workflow, params, log)

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    NAMED WORKFLOW FOR PIPELINE
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { METHYLSEQ } from './workflows/methylseq'

//
// WORKFLOW: Run main nf-core/methylseq analysis pipeline
//
workflow NFCORE_METHYLSEQ {
    METHYLSEQ ()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    RUN ALL WORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//
// WORKFLOW: Execute a single named workflow for the pipeline
// See: https://github.com/nf-core/rnaseq/issues/619
//
workflow {
    NFCORE_METHYLSEQ ()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
