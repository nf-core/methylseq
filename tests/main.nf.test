import com.askimed.nf.test.util.FileUtil;

nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "methylseq"

    test("METHYLSEQ Workflow main.nf") {

        when {
            params {
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            // Assert Workflow Success
            assert workflow.success

            // Bismark Methylation Calls Splitting Report
            assert snapshot(path("$outputDir/bismark/methylation_calls/splitting_report/Ecoli_10K_methylated_1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub1_trimmed_bismark_bt2.deduplicated_splitting_report.txt"),
                            path("$outputDir/bismark/methylation_calls/splitting_report/SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt")).match("splitting_report")

            // Bismark Summary Report
            assert snapshot(path("$outputDir/bismark/summary/bismark_summary_report.txt")).match("summary")

            // Reference Genome
            assert snapshot(path("$outputDir/bismark/reference_genome/BismarkIndex/genome.fa")).match("reference_genome")

            // Qualimap Report
            assert snapshot(path("$outputDir/qualimap/Ecoli_10K_methylated/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub1/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub2/genome_results.txt")).match("qualimap")

            // Software Versions
            assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions")
        }
    }

}

