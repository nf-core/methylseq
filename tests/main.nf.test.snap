{
    "summary": {
        "content": [
            "bismark_summary_report.txt:md5,778a69ca99a0cce3772c78c383984f41"
        ],
        "timestamp": "2023-04-04T08:48:34+0000"
    },
    "reference_genome": {
        "content": [
            "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
        ],
        "timestamp": "2023-04-04T08:48:34+0000"
    },
    "qualimap": {
        "content": [
            "genome_results.txt:md5,7886cc1a96ee892e49898b0d69ed66f2",
            "genome_results.txt:md5,79287f69ff6c8a308f0bb988c97f0e15",
            "genome_results.txt:md5,29e911b75d7160e964530e9607110164"
        ],
        "timestamp": "2023-04-04T08:48:34+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,3e2ad14fb81323eb12880a9050780621"
        ],
        "timestamp": "2023-04-04T08:48:34+0000"
    },
    "splitting_report": {
        "content": [
            "Ecoli_10K_methylated_1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt:md5,292c73be0dfaad92ee7d54ec3e5d0dd6",
            "SRR389222_sub1_trimmed_bismark_bt2.deduplicated_splitting_report.txt:md5,d70ee229a29b68db8634fa7d568b7c53",
            "SRR389222_sub2_trimmed_bismark_bt2.deduplicated_splitting_report.txt:md5,3038a90a6e513c3b1c0de540439eb101"
        ],
        "timestamp": "2023-04-04T08:48:34+0000"
    }
}